### PR TITLE
Basic custom-styles audio player

### PR DESCRIPTION
**Pull request by tomhazledine**
Homepage:
Screenshot 2020-05-07 at 21 54 16

Episode page:
Screenshot 2020-05-07 at 21 54 27

### CODE REVIEW

**Comment by tomhazledine** [comment]
Worth running locally to test it works properly (and not just on my machine!)

`npm run build_dev` will build the assets you need (and `npm run 11ty` as usual to be able to view the pages)

File: src/js/components/Player.js [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        // audio.addEventListener("seeked", handleSeeked, false);
+        // audio.addEventListener("seeking", handleSeeking, false);
+        // audio.addEventListener("stalled", handleStalled, false);
+        // audio.addEventListener("suspend", handleSuspend, false);
+        audio.addEventListener("timeupdate", handleTimeupdate, false);
+        // audio.addEventListener("volumechange", handleVolumechange, false);
+        // audio.addEventListener("waiting", handleWaiting, false);
+    };
+
+    // const handleAbort = (e) => {
```
Comment (edhaz) [inline src/js/components/Player.js:63]: Do we need all this commented out code hanging around?

File: src/layouts/episode.njk [hunk 2]
```diff
@@ -1,61 +1,42 @@
...
-        <time datetime="{{date | datefull }}" itemprop="datePublished">{{date | date}}</time>
+    <div class="meta">
         <span class="visuallyhidden">Modified on <time datetime="{{date | datefull }}" itemprop="dateModified">{{date | date}}</time></span>
         <span class="visuallyhidden" itemprop="author" itemscope="" itemtype="http://schema.org/Person">
             <span itemprop="name"><a href="https://plus.google.com/111879829548102811838" itemprop="url" rel="author">Tom Hazledine</a></span>
         </span>
         <span class="visuallyhidden" itemprop="publisher" itemscope="" itemtype="http://schema.org/Organization">
-            <meta itemprop="name" content="Tom Hazledine" />
+            <meta itemprop="name" content="{{global.siteTitle}}" />
             <div itemprop="logo" itemScope itemtype="https://schema.org/ImageObject">
```
Comment (edhaz) [inline src/layouts/episode.njk:22]: What does this `itemtype` do?

File: src/scss/_components.loading.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
+.icon {
+    width: 2em;
+    height: 2em;
+    fill: $black;
+}
+
+// .icon--loading {
```
Comment (edhaz) [inline src/scss/_components.loading.scss:7]: Ditto with all the comments here.

File: webpack.config.js [hunk 4]
```diff
@@ -6,25 +6,26 @@ const UglifyJsPlugin = require("uglifyjs-webpack-plugin");
...
-        filename: `${AppName}.[name].js`
+        filename: `${AppName}.[name].js`,
     },
     optimization: {
         minimizer: [
             new UglifyJsPlugin({
                 cache: true,
                 parallel: true,
-                sourceMap: true // set to true if you want JS source maps
+                sourceMap: true, // set to true if you want JS source maps
```
Comment (edhaz) [inline webpack.config.js:25]: Unnecessary comment? Seems like `sourceMap: true` explains itself 😄

File: webpack.config.js [hunk 5]
```diff
@@ -47,15 +48,15 @@ module.exports = (env, argv) => ({
...
                                 plugins: ["line-numbers"],
                                 theme: "default", //"coy","tomorrow"
-                                css: true
-                            }
-                        ]
-                    ]
-                }
+                                css: true,
+                            },
+                        ],
```
Comment (edhaz) [inline webpack.config.js:57]: Some new comma madness? 😆

File: src/layouts/episode.njk (hunk 2 above)
Comment (tomhazledine) [reply src/layouts/episode.njk:22]: Part of the structured schema data - really helpful for SEO etc. https://schema.org/

File: src/js/components/Player.js (hunk 1 above)
Comment (tomhazledine) [reply src/js/components/Player.js:63]: Not "best practice" but the comments are really useful for me here (still in-progress on the enhanced player view).

File: webpack.config.js (hunk 5 above)
Comment (tomhazledine) [reply webpack.config.js:57]: Eurgh - that's my auto-formater. Prettier changed its default settings. Work prefers trailing commas (makes diffs cleaner going forward), so they're not in my Prettier overrides.

Thoughts? Keep trailing commas as standard?

**Comment by tomhazledine** [comment]
Good point about the top-margin on the episode page - I'll tweak it.

File: webpack.config.js (hunk 5 above)
Comment (edhaz) [reply webpack.config.js:57]: Sure, I'm happy to keep them

File: src/layouts/episode.njk (hunk 2 above)
Comment (edhaz) [reply src/layouts/episode.njk:22]: 👍

File: src/js/components/Player.js (hunk 1 above)
Comment (edhaz) [reply src/js/components/Player.js:63]: 👍

**Pull request merged**